lib/test_case_manager_filter.rb

```diff
@@ -40,9 +40,7 @@ def done
       end
 
       config.on_event :test_case_finished do |event|
-        unless event.result.skipped?
-          tc_manager.signal(:end_case, event)
-        end
+        tc_manager.signal(:end_case, event)
```

Comment:
Let's assume we have two test cases, caseA and caseB. caseA has `skip_this_scenario` in one of its steps due to conditions can not met (e.g, scenario requires at least two nodes, but we are on SNO, which has only one node).
When starting the execution on caseB after caseA is skipped, Cucumber::Core::Test::Case is set to caseB, and BushSlicer::PolarShift::ScenarioWrapper still points to caseA, which cause logic error in `start_scenario_for`, and caseB is skipped due to the logic error/mismatch.
If we have caseC, caseD, ... followed, it will be skipped, too, due to the same logic error/mismatch.

The change here removed the skip check, so when caseA is skipped, we send a signal to indicate we have reached the end of case of caseA, so Cucumber::Core::Test::Case and BushSlicer::PolarShift::ScenarioWrapper match again.